R/query-string.R

```diff
@@ -8,29 +8,36 @@ queryStringFilter <- function(req){
 #' @importFrom utils URLdecode
 #' @noRd
 parseQS <- function(qs){
-  if (is.null(qs) || length(qs) == 0 || qs == ""){
+  # Is there data in the request?
+  if (is.null(qs) || length(qs) == 0 || qs == "") {
     return(list())
   }
-  if (stri_startswith_fixed(qs, "?")){
-    qs <- substr(qs, 2, nchar(qs))
-  }
-
-  qs <- gsub("+", " ", qs, fixed=TRUE)
+ # Is it JSON data?
+  if (stri_startswith_fixed(qs, "{")) {
```

Comment:
The original motivation for this feature was to support browser `<form>`s which tend to use the key/value (querystring) format in a POST format. I think it's important to maintain support for that so browser forms can easily get data in. I do agree that most APIs would expect JSON, though.